### DISCUSSION TITLE
How to use @wdio/allure-reporter with Cucumber framework?

### DISCUSSION THREAD

**Discussion by Finiodor** [webdriverio/webdriverio#7455] [Q&A]
Actually, i'm interested in the **addStory(storyName)** feature. I see examples for Mocha but have no idea how to use package with CucumberJS. I know that there is package **allure-cucumberjs** but maybe you have any tips on how to set story to reporter via @wdio/allure-reporter ?

**Comment by christian-bromann** [comment]
@finelord how is the [current documentation](https://webdriver.io/docs/allure-reporter#usage) not helping you?

**Reply by finelord** [reply]
I believe you mentioned me by mistake?

**Reply by christian-bromann** [reply]
sorry, yes

@Finiodor

**Reply by Finiodor** [reply]
I see an example for Mocha but not for Cucumber. I don't have describe=>it structure in the .feature files in cucumber.

**Reply by christian-bromann** [reply]
You have step definition instead where you need to import this.

**Reply by Finiodor** [reply]
Great, thank you. Please, add example for Cucumber as well to the documentation.

**Reply by christian-bromann** [reply]
Feel free to send a PR to the project. We always appreciate docs improvements.

**Reply by anupamck** [reply]
@Finiodor  - Were you able to figure this out for Cucumber? I am stuck with this problem as well!

**Reply by Finiodor** [reply]
> @Finiodor - Were you able to figure this out for Cucumber? I am stuck with this problem as well!

Yes, it started working after adding them at the step's top, like this:

```
Given('My step descriptions', async () => {
    allureReporter.addFeature('Feature_name');
    allureReporter.addStory('Story_name');
    await myStepFunc();
})
```

**Reply by anupamck** [reply]
Thanks @Finiodor - This worked and I included it as part of the documentation too.
